helper-cli/src/funTest/kotlin/commands/CreateAnalyzerResultFromPackageListCommandFunTest.kt

```diff
@@ -22,21 +22,21 @@ package org.ossreviewtoolkit.helper.commands
 import com.github.ajalt.clikt.testing.test
 
 import io.kotest.core.spec.style.WordSpec
+import io.kotest.engine.spec.tempdir
```

Comment:
Could the remaining two places also be migrated?
* `scanner/src/test/kotlin/ScannerTest.kt`
* `scanner/src/test/kotlin/utils/FileListResolverTest.kt`

Reply:
I had investigated these already, and they should stay as-is:

In `ScannerTest.kt`, while the `FakeProvenanceDownloader` deliberately dose not delete the temporary directory, the wrapping scanner does.

And in `FileListResolverTest.kt`, there is an explicit test that the directory already gets deleted, see

https://github.com/oss-review-toolkit/ort/blob/48ae81634c5a3bd4d55e89d2474dc05825a5ae03/scanner/src/test/kotlin/utils/FileListResolverTest.kt#L64-L80

Reply:
I'm not sure if I follow the rationale. Which of the following do you mean?

1. migration is not needed because production code takes care of deleting
2. migration is not possible because it would break something or invalidate the test case.

In case of 1. should we migrate anyway, just for consistency?

Reply:
It's 1., plus deletion would defeat the purpose of the test (though not cause an error), so IMO these should stay as-is.

Reply:
Why would deletion defeat the purpose of the test? To my understanding the kotest function would delete the dir only after the spec has been entirely executed and assertion about deletion has been made ?

Reply:
Ok, "defeat the purpose of the test" probably was the wrong wording.

My point for `FakeProvenanceDownloader` is that it should be implemented like any other "real" `ProvenanceDownloader` and not make use of test-only functions as part of its implementation. `tempdir()` cannot be called from `FakeProvenanceDownloader` anyway as the function requires `TestConfiguration` context. And `ProvenanceDownloader.download()` documents "The caller is responsible for deleting the directory", so it really should not delete the directory itself.

For `createTempDirWithFiles()` in `FileListResolverTest` there would be no harm in using `tempdir()`, but it's completely unnecessary as there already is `dir shouldNotBe aDirectory()` as part of the test.

Reply:
> For `createTempDirWithFiles()` in `FileListResolverTest` there would be no harm in using `tempdir()`

Ok, I did that now just for consistency, like you requested.